api/types/filters/parse.go

```diff
@@ -99,7 +100,7 @@ func FromJSON(p string) (Args, error) {
 	// Fallback to parsing arguments in the legacy slice format
 	deprecated := map[string][]string{}
 	if legacyErr := json.Unmarshal(raw, &deprecated); legacyErr != nil {
-		return args, invalidFilter{errors.Wrap(err, "invalid filter")}
+		return args, errdefs.InvalidParameter(errors.Wrap(err, "invalid filter"))
```

Comment:
We should probably try to avoid importing `errdefs` only for this one; could this be handled by `invalidFilter` (i.e., don't return name or value if they're empty?)

Reply:
Good point, done